src/NuGet.Core/NuGet.Build.Tasks.Console/Program.cs

```diff
@@ -76,6 +76,16 @@ public static async Task<int> Main(string[] args)
             }
 #endif
 
+            // Check whether the ask is to generate the restore graph file.
+            if (MSBuildStaticGraphRestore.IsOptionTrue("GenerateRestoreGraphFile", arguments.Options))
+            {
+                using (var dependencyGraphSpecGenerator = new MSBuildStaticGraphRestore(debug: debug))
+                {
+                    return dependencyGraphSpecGenerator.WriteDependencyGraphSpec(arguments.EntryProjectFilePath, arguments.MSBuildGlobalProperties, arguments.Options) ? 0 : 1;
+                }
+            }
+
+            // Otherwise run restore!
```

Comment:
Since this exe ships with our targets, I think we should change restore to take `restore` as the first arg (and `GenerateRestoreGraphFile` can be changed to be the first arg), to make sure there is no chance of conflict, and to avoid someone trying to do a restore and generate restore graph file in the same execution. Plus it fits more in line with how most multi-command programs, like the dotnet cli or nuget.exe, work.

Reply:
I actually had decided against changing it because I didn't want to change the expected arguments of the exe.

I can add the extra comment if you still think so. 

This is not a traditional exe at this point, so I don't see that as a super convincing argument.